hudi-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -193,15 +194,17 @@ private void assignInserts(WorkloadProfile profile, JavaSparkContext jsc) {
         }
 
         // Go over all such buckets, and assign weights as per amount of incoming inserts.
-        List<InsertBucket> insertBuckets = new ArrayList<>();
+        List<Pair<InsertBucket, Double>> insertBuckets = new ArrayList<>();
```

Comment:
we might as well store InsertBucket2CumulativeWeight in this list since you have a class.